pkg/performanceprofile/controller/performanceprofile/hypershift/hypershift.go

```diff
@@ -25,6 +25,14 @@ const (
 	HostedClustersNamespaceName = "clusters"
 )
 
+type counter map[string]int
```

Comment:
This commit should not go in and it's only meant for the reviewers sake, to provide some rough numbers about  how many time each CRUD operation is being called.

I'll provide some numbers soon.

Reply:
Some statistics about client calls:
When some owned object is changing:
```
I0630 10:28:19.469666       1 performanceprofile_controller.go:537] Client counter END of reconciliation:
 FunctionName="Get", Called 2 times
FunctionName="GetFromConfigMap", Called 3 times
```
 
When owned objects are creating:
```
I0630 15:40:13.183077       1 performanceprofile_controller.go:424] Client counter BEGIN of reconciliation:
 FunctionName="Get", Called 1 times
...
I0630 15:40:13.313975       1 performanceprofile_controller.go:425] Client counter END of reconciliation:
 FunctionName="Get", Called 5 times
FunctionName="GetFromConfigMap", Called 3 times
FunctionName="Create", Called 3 times

```

Let's analyze the numbers:
1. When a change being detected (meaning some of the objects has been tampered) 
standard `Get` being called 2 times and `GetFromConfigMap` , which is more costly because it involves with listing `ConfigMap`s a and deserialize   their data, called 3 times.

2. When objects are create (fresh performanceProfilie applied) 
standard `Get` being called 4 times and `GetFromConfigMap` called 3 times and `Create` called 3 times.

3. When nothing is changing, no calls to the API server are being made by the controller as expected

According to the above numbers and considering the tight timelines we have, I think the provided solution for fixing the client implementation in this PR is good enough.

In case of performance degradation/additional bugs/gaps that were not covered, we can re evaluate in the future. 
But I this point in time I believe we should merge quickly in order to unlift the blocker from the testing development.

Reply:
I agree with your assessment given the numbers we have. I'll have another review ASAP.